inf-ruby.el

```diff
@@ -683,6 +675,48 @@ This function also removes itself from `pre-command-hook'."
                 (point)))))
         (buffer-substring-no-properties (point) (line-end-position))))))
 
+(defun ruby-shell--encode-string (string)
+  "Escape all backslashes, single quotes, and newlines in STRING."
+  (cl-reduce (lambda (string subst)
+               (replace-regexp-in-string (car subst) (cdr subst) string))
+             '(("\\\\" . "\\\\\\\\")
+               ("'" . "\\\\'")
+               ("\n" . "'\"\\\\n\"'"))
+             :initial-value string))
+
+(defun ruby-send-string (string &optional file line)
+  "Send STRING to the inferior Ruby process."
+  (interactive
+   (list (read-string "Ruby command: ") nil t))
+  (let* ((file-and-lineno (concat (when file
+                                    (format ", %S" (inf-ruby-file-local-name file)))
+                                  (when (and file line)
+                                    (format ", %d" line))))
+         (code (format "eval('%s', %s%s)\n"
+                       (ruby-shell--encode-string string)
+                       inf-ruby-eval-binding
+                       file-and-lineno)))
+    (if (or (null (process-tty-name (inf-ruby-proc)))
+            (<= (string-bytes code)
+                (or (bound-and-true-p comint-max-line-length)
+                    1024))) ;; For Emacs < 28
+        (comint-send-string (inf-ruby-proc) code)
+      (let* ((temporary-file-directory
+              (if (file-remote-p default-directory)
+                  (concat (file-remote-p default-directory) "/tmp")
```

Comment:
Let's try simplifying this as

```el
(let* ((temporary-file-directory (temporary-file-directory))
       (tempfile (make-temp-file "irb"))))
```

Then it'll also work on Windows remote hosts, I guess. The minimum versions will need to be bumped to 26.1, but that's been a long time coming.

Reply:
nice, I just had copied it from python-mode but that function does exactly what we want in 3 lines less, and better as it's not hardcoding /tmp